ethreceipts/ethreceipts.go

```diff
@@ -818,7 +818,10 @@ func collectOk[T any](in []T, oks []bool, okCond bool) []T {
 // }
 
 func groupLogsByTransaction(logs []types.Log, numTxns int) [][]*types.Log {
-	out := make([][]*types.Log, numTxns)
+	if numTxns == 0 {
```

Comment:
maybe we should just remove numTxns argument here and change this with.. `if len(logs) == 0 {`